dropshot/tests/test_tls.rs

```diff
@@ -230,20 +231,20 @@ async fn test_tls_refresh_certificates() {
             > {
                 // Verify we're seeing the right cert chain from the server
                 if *end_entity != certs[0] {
-                    return Err(rustls::Error::InvalidCertificateData(
-                        "Invalid end cert".to_string(),
+                    return Err(rustls::Error::InvalidCertificate(
+                        rustls::CertificateError::BadEncoding,
```

Comment:
Github won't let me link to it, but we're using: https://docs.rs/rustls/0.21.0/rustls/struct.ConfigBuilder.html#method.with_custom_certificate_verifier

Which is where this `make_cert_verifier` closure actually gets used - as the name implies, it just lets us inject whatever validation we want for our tests.

Is `BadEncoding` the right error here? Honestly, it doesn't matter - we just need to throw *some* error, and confirm that we see failures during the handshake below.

Arguably, since we're checking for an *exact* certificate match, this is an unrealistically picky verifier, so *maybe* `rustls::CertificateError::Other` is the most accurate? But again, I don't think it matters too much.

Reply:
With `Other` I'd have to give it an `impl Error`... it seemed not worth it. Which is how I landed on `InvalidCertificate`